tests/features/steps/secure-files.py

```diff
@@ -14,21 +15,109 @@
 @when('The consumer publishes secure file with a valid token')
 def step_impl(context):
     params = (
-        ('id', res[2]),
-        ('token', tokens["2_5_write"]),
+        ('id', res[8]),
+        ('token', tokens["8_10_rw"]),
+    )
+    files = {
+        'file': ('sample.txt', open('sample.txt', 'rb')),
+        'metadata': ('meta.json', open('meta.json', 'rb')),
+    }
+    post_files(params, files, context)
+
+
+@when('The consumer publishes with a valid token(1)')
+def step_impl(context):
+    params = (
+
+        ("id", res[9]),
+        ('token', tokens["8_10_rw"]),
+
+    )
+    files = {
+        'file': ('sample.txt', open('sample.txt', 'rb')),
+        'metadata': ('meta.json', open('meta.json', 'rb')),
+    }
+    post_files(params, files, context)
+
+
+@when('The consumer publishes with a valid token(2)')
+def step_impl(context):
+    params = (
+
+        ("id", res[10]),
+        ('token', tokens["8_10_rw"]),
+
+    )
+    files = {
+        'file': ('sample.txt', open('sample.txt', 'rb')),
+        'metadata': ('meta.json', open('meta.json', 'rb')),
+    }
+    post_files(params, files, context)
+
+
+@when('The consumer publishes with a valid token(3)')
+def step_impl(context):
+    params = (
+
+        ("id", res[11]),
+        ('token', tokens["11_rw"]),
+
     )
     files = {
         'file': ('sample.txt', open('sample.txt', 'rb')),
         'metadata': ('meta.json', open('meta.json', 'rb')),
     }
-    check_pub_file(params, files, context)
+    post_files(params, files, context)
+
+
+@when('The consumer downloads file by passing multiple resource ids and a token')
+def step_impl(context):
+    url = 'https://localhost/download?token=' + tokens["8_10_rw"] + '&id=' + res[8] + ',' + res[10]
```

Comment:
Use uri params in the request module. Do not hardcode this.

Reply:
made changes